### PR TITLE
Env config

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
My Version which replaces config by environment Variables. Variables are loaded into the 'hypes' file. They are therefore accessible throughout the framework.  This commit is redundant to some work done in #13. We should coordinate better next time.

### CODE REVIEW

**Comment by MartinThoma** [comment]
@MarvinTeichmann I think this pull request is superseeded by mine. Why did you start it when mine was open for 3 days now?

However, you seem to have edited quite a bit of code which is not related to the environment variable configuration. So after merging my pull request, you might want to do those changes in another pull request. I think this one should be closed.

**Comment by MarvinTeichmann** [comment]
Inertially this was planned as an update for your pull request. It build on top of commit 3a93c89b8b5823353d9978a407e54676029f57c1 and was going to include this into your pull request. However before I committed my last changes you did 17b888a953cf10c9a770b41962ee657d360e94ce which has similar functionality then 66e3f02cf09caf35264faf516ea4095a0e0138f1

**Comment by MarvinTeichmann** [comment]
I might be a bit biased, but when I compare 17b888a and 66e3f02 I prefer mine right now. On that way all relevant configuration information are eventually stored in 'hypes'. This seems to be a bit cleaner.

File: README.md [hunk 1]
```diff
@@ -25,6 +25,21 @@ Flags:
...
+
+TensorVision is configured with environment variables. It is quite easy to
+set them yourself (see [multiple ways](http://unix.stackexchange.com/a/117470/4784)).
+The supported variables are:
+
+* `TV_STEP_STR`: Set what you want to see each 100th step of the training.
+  The default is `'Step {step}/{total_steps}: loss = {loss_value:.2f} ( {sec_per_batch:.3f} sec (per Batch); {examples_per_sec:.1f} examples/sec;)'`
+* `TV_IS_DEV`: Either 0 or 1 - set if you want to see debug messages.
+* `TV_DIR_RUNS`: Directory where runs are saved. If relative it is processed relative to hypes.
+* 'TV_DIR_DATA': Directory where data is loaded from. If relative it is processed relative to hypes.
```
Comment (MartinThoma) [inline README.md:41]: backticks, not `'`

File: bin/tv-train [hunk 2]
```diff
@@ -47,26 +51,26 @@ def _copy_parameters_to_traindir(input_file, target_name, target_dir):
         directory where training data is saved
     """
     target_file = os.path.join(target_dir, target_name)
+    input_file = os.path.join(hypes['dirs']['base_path'], input_file)
```
Comment (MartinThoma) [inline bin/tv-train:54]: What exactly is `base_path`?

File: bin/tv-train [hunk 3]
```diff
@@ -27,13 +27,17 @@ FLAGS = flags.FLAGS
...
+                        level=logging.INFO,
+                        stream=sys.stdout)
 
-logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s',
-                    level=logging.INFO,
-                    stream=sys.stdout)
 
-
-def _copy_parameters_to_traindir(input_file, target_name, target_dir):
+def _copy_parameters_to_traindir(hypes, input_file, target_name, target_dir):
```
Comment (MartinThoma) [inline bin/tv-train:40]: Isn't `input_file` also in `hypes`? Shouldn't we - now that we change the function signature anyway - make it simplyer then?

File: bin/tv-train [hunk 4]
```diff
@@ -76,32 +80,33 @@ def initialize_training_folder(hypes, train_dir):
...
     # TODO: read more about loggers and make file logging neater.
 
-    hypes_file = tf.app.flags.FLAGS.hypes
-    _copy_parameters_to_traindir(hypes_file, "hypes.json", target_dir)
+    hypes_file = os.path.basename(tf.app.flags.FLAGS.hypes)
+    _copy_parameters_to_traindir(
+        hypes, hypes_file, "hypes.json", target_dir)
     _copy_parameters_to_traindir(
-        hypes['model']['input_file'], "data_input.py", target_dir)
+        hypes, hypes['model']['input_file'], "data_input.py", target_dir)
```
Comment (MartinThoma) [inline bin/tv-train:87]: `hypes['model']['input_file'], hypes['model']['architecture_file'], hypes['model']['objective_file']` and `hypes['model']['optimizer_file']` always exist in `hypes`. We can therefor only pass `hypes`.

**Comment by MartinThoma** [comment]
@MarvinTeichmann What my pull request has and yours (by now) doesn't have:
- documentation (README.md - for all environment variables)
- tests (although almost non-existant)
- loading of plugins

You could add those to your pull request simply by copy-paste. 

Otherwise, I agree. Having all configuration in one place is nice.

**Comment by MarvinTeichmann** [comment]
I think what will be easiest is first apply your pull request and then cherry-pick the commits I have done. I will do this now.

File: bin/tv-train (hunk 4 above)
Comment (MarvinTeichmann) [inline bin/tv-train:87]: Yes, they do always exist in hypes. So you to propose to only pass the string, e.g. 'optimizer_file'?

**Comment by MarvinTeichmann** [comment]
Incooperated in #20

File: bin/tv-train (hunk 2 above)
Comment (MarvinTeichmann) [inline bin/tv-train:54]: `base_path = os.path.dirname(os.path.realpath(hypes_fname))` Also der Pfad zu der hypes datei. Bzgl. dieses pfades sollen alle relativen pfade interpretiert werden.

**Pull request closed without merging**